application/src/main/java/org/thingsboard/server/service/subscription/DefaultTbLocalSubscriptionService.java

```diff
@@ -387,30 +421,30 @@ private void processSubscriptionData(UUID entityId,
         callback.onSuccess();
     }
 
-    private void modifySubscription(TenantId tenantId, EntityId entityId, TbSubscription<?> subscription, boolean add) {
+    private ModifySubscriptionResult modifySubscription(TenantId tenantId, EntityId entityId, TbSubscription<?> subscription, boolean add) {
         TbSubscription<?> missedUpdatesCandidate = null;
-        TbEntitySubEvent event;
-        subsLock.lock();
-        try {
-            TbEntityLocalSubsInfo entitySubs = subscriptionsByEntityId.computeIfAbsent(entityId.getId(), id -> new TbEntityLocalSubsInfo(tenantId, entityId));
-            event = add ? entitySubs.add(subscription) : entitySubs.remove(subscription);
-            if (entitySubs.isEmpty()) {
-                subscriptionsByEntityId.remove(entityId.getId());
-                entityUpdates.remove(entityId.getId());
-            } else if (add) {
-                missedUpdatesCandidate = entitySubs.registerPendingSubscription(subscription, event);
-            }
-        } finally {
-            subsLock.unlock();
+        TbEntityLocalSubsInfo entitySubs = subscriptionsByEntityId.computeIfAbsent(entityId.getId(), id -> new TbEntityLocalSubsInfo(tenantId, entityId));
+        TbEntitySubEvent event = add ? entitySubs.add(subscription) : entitySubs.remove(subscription);
+        if (entitySubs.isEmpty()) {
+            subscriptionsByEntityId.remove(entityId.getId());
+            entityUpdates.remove(entityId.getId());
+        } else if (add) {
+            missedUpdatesCandidate = entitySubs.registerPendingSubscription(subscription, event);
         }
+        return new ModifySubscriptionResult(tenantId, entityId, subscription, missedUpdatesCandidate, event);
+    }
+
+    private void pushSubscriptionEvent(ModifySubscriptionResult modifySubResult) {
+        TbEntitySubEvent event = modifySubResult.getEvent();
```

Comment:
try catch with logging (warn) for both pushSubscriptionEvent and modifySubscription